avocado/utils/software_manager/backends/yum.py

```diff
@@ -58,6 +52,17 @@ def repo_config_parser(self):
             self._cfgparser.read(self.REPO_FILE_PATH)
         return self._cfgparser
 
+    @property
+    def yum_base(self):
+        if self._yum_base is None:
+            if HAS_YUM_MODULE:
+                self._yum_base = yum.YumBase()
+            else:
+                log.debug("%s module for Python is required to use the "
+                          "'provides' command. Using the basic support "
+                          "from rpm and %s commands", self.cmd, self.cmd)
+        return self._yum_base
+
```

Comment:
Since your constant `HAS_YUM_MODULE` it already a hint, what about?

```python
if not HAS_YUM_MODULE:
    log.debug("%s module for Python is required to use the "
              "'provides' command. Using the basic support"
              "from rpm and %s commands", self.cmd, self.cmd)
    return None
return self._yum_base or yum.YumBase()
```

Reply:
Just realized that the last line must update `_yum_base`, but I think that you got the idea. To invert the logic and log earlier.

Reply:
@beraldoleal do you mean logging at the module import time?  I honestly think that if you're following the logs, it'd be better to have that message closer to the actual usage, instead of much earlier.

Reply:
No, but here on this property checking the HAS_YUM_MODULE first. Just a a minor change, probably will get us to the same result, so don't mind here.

Reply:
OK... I'll merge it as is then.  Thanks!